src/test/java/io/reactivex/internal/operators/flowable/FlowableSwitchTest.java

```diff
@@ -1202,4 +1203,34 @@ public Object apply(Integer w) throws Exception {
         .assertNoErrors()
         .assertComplete();
     }
+
+    @Test
+    public void undeliverableUponCancel() {
```

Comment:
I see that all `undeliverableUponCancel()` test methods in different test classes are quite similar. Do you think, does it make sense to extract this code, parametrize it and move it to the newly created `abstract class SwitchTest`, which these classes can extend to avoid code duplication? I haven't checked if it's doable.

Reply:
Try it, you'll see.